### ISSUE TITLE
SiLU/Swish returns incorrect gradients for broadcasted Tensor beta

### ISSUE THREAD

**Issue by twelfthlabor** [tensorflow/tensorflow#127690]
### Issue type
Bug

### Have you reproduced the bug with TensorFlow Nightly?
No. Reproduced with TensorFlow 2.21.0 and separately with `nn_impl.py` from master at `08088f1035ad1f5fdbc8d0e41dd72dfc06dc522c` loaded against the 2.21.0 CPU kernels. This was not a full source build.

### Source
binary

### TensorFlow version
2.21.0; also current master Python implementation as described above

### Custom code
Yes

### OS platform and distribution
macOS 26.5.1, arm64

### Python version
3.11.5

### Bazel / compiler / CUDA / GPU
Not used; CPU execution

### Current behavior?
`tf.nn.silu` (also exported as `tf.nn.swish`) accepts a Tensor `beta` and evaluates `features * sigmoid(beta * features)`, but its custom gradient reduces the beta derivative to a scalar regardless of beta's shape. For a per-column beta, this combines unrelated parameter derivatives. When features itself broadcasts to beta, its gradient is also returned with the expanded output shape rather than the input shape.

For the example below, the beta gradient should have shape `(2,)` and values approximately `[0.6032018708, 0.0760146276]`. Instead it has shape `()` and value `0.6792164984`. The ordinary TensorFlow expression produces the expected gradient.

The cause is the unconditional `reduce_sum` for beta and the missing broadcast reduction for features in the custom gradient. I have a focused fix and finite-difference regression tests for both inputs, including dynamic shapes, and will submit a PR.

### Standalone code to reproduce the issue
```python
import tensorflow as tf

x = tf.constant([[1., 2.], [3., 4.]], tf.float64)
beta = tf.constant([1., 2.], tf.float64)

with tf.GradientTape() as tape:
    tape.watch(beta)
    y = tf.nn.silu(x, beta=beta)
actual = tape.gradient(y, beta)

with tf.GradientTape() as tape:
    tape.watch(beta)
    y = x * tf.sigmoid(beta * x)
expected = tape.gradient(y, beta)

print("actual:", actual.numpy(), actual.shape)
print("expected:", expected.numpy(), expected.shape)
assert actual.shape == expected.shape
```

### Relevant log output
```text
actual: 0.6792164984049589 ()
expected: [0.60320187 0.07601463] (2,)
AssertionError
```

**Comment by Kayyuri** [comment]
Hi @twelfthlabor, Thank you for reporting this issue!

I see you have already taken the initiative and submitted a [PR](https://github.com/tensorflow/tensorflow/pull/127691) to fix this issue. Once PR is merged, the issue will be resolved.

Thanks again for your patience and for contributing a fix to TensorFlow!
